Zabolotskiy/hash/main.cpp

```diff
@@ -0,0 +1,64 @@
+#include <Windows.h>
+#include <fstream>
+#include "../src/hashlibpp.h"
+using namespace std;
+
+HANDLE inputMutex, outputMutex;
+ifstream *infile;
+ofstream *outfile;
+hashwrapper *md5 = new md5wrapper();
+hashwrapper *sha1 = new sha1wrapper();
+char* block;
+int sizeBlock = 0;
+char* hashType;
+
+void Thread(void* pParams)
+{
+		block = new char[sizeBlock];
+		while (!infile->eof())
+		{
+			WaitForSingleObject(inputMutex, INFINITE);
+				infile->read((char*)(block), sizeBlock);
+			ReleaseMutex(inputMutex);
+			
+			WaitForSingleObject(outputMutex, INFINITE);
+				if (!strcmp(hashType, "md5"))	*outfile << md5->getHashFromString(block) << endl;
+				else	*outfile << sha1->getHashFromString(block) << endl;
```

Comment:
Didn't notice at first - If your program is run with, for example, `crc32` parameter, it will calculate sha1 hash (because `strcmp` says that `"md5" != "crc32"`, and will save it to `.hashes_crc32` file. This is confusing.